### PR TITLE
Fix download bugs, add subject to download structure, support BIDS uploads

### LINKED ISSUE #421: Download tweaks and bug fixes
- [x]  Make BIDS download format the standard (Group -> Project -> Subject -> Session -> Acquisition)
 - [x] Add query param to set base folder name. Default to `scitran`
 - [x] Fix bug in path format for sessions with timestamps and no label

### PR DESCRIPTION
**Pull request by nagem**
Closes #421.

### CODE REVIEW

**Comment by ryansanford** [comment]
@nagem No one is assigned to this PR. Can you clarify the status, and assign as needed? Thanks.

**Comment by kofalt** [comment]
> hierarchy.py

hierarchy py

File: api/download.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def _bulk_preflight_archivestream(self, file_refs):
                 file_cnt += 1
 
         if len(targets) > 0:
-            filename = arc_prefix + datetime.datetime.utcnow().strftime('%Y%m%d_%H%M%S') + '.tar'
+            filename = arc_prefix + '_ '+datetime.datetime.utcnow().strftime('%Y%m%d_%H%M%S') + '.tar'
```
Comment (kofalt) [inline api/download.py]: Why the new underscore?

File: api/placer.py [hunk 2]
```diff
@@ -161,13 +161,22 @@ def process_file_field(self, field, info):
...
+        if container.level != 'subject':
+            self.container_type = container.level
+            self.id_            = container.id_
+            self.container      = container.container
+            self.save_file(field, info)
+        else:
+            if field is not None:
+                files.move_form_file_field_into_cas(field)
+            if info is not None:
+                container.upsert_file(info)
```
Comment (kofalt) [inline api/placer.py]: Is it okay that you don't call `save_file` in this route?

File: api/download.py (hunk 1 above)
Comment (nagem) [inline api/download.py]: So it is in the same format as the [other download path](https://github.com/scitran/core/pull/446/files#diff-5f6b428f8ab6dc5134ff7b9cbcad2062R224).

File: api/placer.py (hunk 2 above)
Comment (nagem) [inline api/placer.py]: I'm doing a modified version of that logic to handle files at the subject level.

**Comment by nagem** [comment]
@dpuccetti FYI this ticket adds a `prefix` query param that allows the requester to set the name of the downloaded zip. 

```
  queryParameters:
    prefix:
      type: string
      required: false
      description: |
       A string to customize the name of the download
       in the format <prefix>_<timestamp>.tar.gz.
       Defaults to "scitran".
```

**Comment by nagem** [comment]
Going to merge this around 12:30 today, let me know if you see anything before then!

**Comment by kofalt** [comment]
Changes look okay to me; did not review `hierarchy.py`.

**Comment by rentzso** [comment]
LGTM

**Pull request merged**